File: Features/Reservations/Resources/Form/ReservationReadResource.cs
using System;
using System.Collections.Generic;
using BlueWaterCruises.Features.Customers;
using BlueWaterCruises.Features.Destinations;
using BlueWaterCruises.Features.Drivers;
using BlueWaterCruises.Features.PickupPoints;
using BlueWaterCruises.Features.Ports;

namespace BlueWaterCruises.Features.Reservations {

    public class ReservationReadResource {

        public Guid ReservationId { get; set; }

        public string Date { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPersons { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public string Remarks { get; set; }

        public CustomerDropdownResource Customer { get; set; }
        public DestinationDropdownResource Destination { get; set; }
        public PickupPointDropdownResource PickupPoint { get; set; }
        public DriverDropdownResource Driver { get; set; }
        public ShipDropdownResource Ship { get; set; }
        public PortDropdownResource Port { get; set; }

        public UserResource User { get; set; }

        public List<PassengerReadResource> Passengers { get; set; }

    }

}

File: ClientApp/src/app/features/reservations/classes/resources/form/reservation/reservation-read-resource.ts
import { Guid } from "guid-typescript"
// Custom
import { CustomerDropdownResource } from "../dropdown/customer-dropdown-resource"
import { DestinationDropdownResource } from '../dropdown/destination-dropdown-resource'
import { DriverDropdownResource } from '../dropdown/driver-dropdown-resource'
import { PassengerReadResource } from "../passenger/passenger-read-resource"
import { PickupPointDropdownResource } from '../dropdown/pickupPoint-dropdown-resource'
import { PortDropdownResource } from '../dropdown/port-dropdown-resource'
import { ShipDropdownResource } from '../dropdown/ship-dropdown-resource'

export class ReservationReadResource {

    reservationId: Guid

    date: string
    email: string
    phones: string
    remarks: string
    adults: number
    kids: number
    free: number
    totalPersons: number
    ticketNo: string

    customer: CustomerDropdownResource
    destination: DestinationDropdownResource
    pickupPoint: PickupPointDropdownResource
    driver: DriverDropdownResource
    ship: ShipDropdownResource
    port: PortDropdownResource

    passengers: PassengerReadResource

}

